### DISCUSSION TITLE
GetHerokuVars.sh

### DISCUSSION THREAD

**Discussion by bjornoleh** [NightscoutFoundation/xDrip#2422] [General]
**Nightscout on Google Cloud**

Out of curiosity, I tested the GetHerokuVars.sh script:

Since MFA is mandatory on Heroku, I needed to create a new authorization token, but this actually worked. However, the script failed, probably because I have more than one app managed by my Heroku user (NS apps from family members managed in my account).

Just an FYI in case feedback is welcome.

`/xDrip/scripts$ bash GetHerokuVars.sh`

```
Logged in as xxxx@gmail.com
ns-xxxx ns-yyy ns-zzz ns-qwerty (anonymised NS app names...)
GetHerokuVars.sh: line 15: heroku_vars.txt: Permission denied
 ›   Error: Unexpected arguments: ns-xxxx, ns-yyy, ns-zzz, ns-qwerty
 ›   See more help with --help
new vars are in the file heroku_vars.txt.
Please look at it, and if all is ok, try copying it to /etc/nsconfig
```

**Comment by tzachi-dar** [comment]
it seems that you need to run it with sudo, as you did not have write permissions to the local directory.

**Reply by bjornoleh** [reply]
Thanks, that seems to have been part of the issue, but run as sudo it still seems to not handle multiple apps on the same account. The script wrote the heroku_vars.txt file this time, but the file was empty. I got the same error as last time:
 `Error: Unexpected arguments: ns-xxxx, ns-yyy, ns-zzz, ns-qwerty`
